File: lua/cycle_b.lua
local M = {}

function M.new(tag)                 return { tag = tag } end
function M.bounce_to_alpha(b)
    local cycle_a = require("cycle_a")         -- closes the cycle
    return cycle_a.describe({ name = "bounce-from-" .. b.tag })
end

return M

File: monorepo_rust/crates/app/src/cycle_b.rs
// C1.b — closes the cycle back to Alpha.

use crate::cycle_a::Alpha;

pub struct Bravo {
    pub tag: String,
}

impl Bravo {
    pub fn new(tag: &str) -> Self { Bravo { tag: tag.into() } }

    pub fn bounce_to_alpha(&self) -> String {
        Alpha::new(&format!("bounce-from-{}", self.tag)).describe()
    }
}
